source/tutorials/kubernetes.rst

```diff
@@ -3,170 +3,340 @@
+#. Run the kubeadm command to initialize the master node with the
+   :command:`--cri-socket` parameter:
 
-Configure and run CRI-O + kata-runtime
-======================================
+   .. important:: 
 
-#.  Enable the CRI-O service:
+      You may need to add additional parameters to the command below,
+      depending the pod network addon in use. In this example, *flannel* is
+      being used for the pod networking. See `Choose a pod network add-on`_
+      for more information.
 
-    .. code-block:: bash
+   .. code-block:: bash
 
-       sudo systemctl enable crio.service
+      sudo kubeadm init \
+      --cri-socket=/run/containerd/containerd.sock \ #required for containerd
+      --pod-network-cidr 10.244.0.0/16   #required for flannel
 
-#.  Enter these commands to restart the system services so that CRI-O will be restarted, and will automatically start with every reboot.
 
-    .. code-block:: bash
+#. (Optional) By default, containerd will use the default runc as the default
+   runtime. The with or without `Kata Containers`_. runtime can be set
+   per-pod. See the Kata Containers documentation on `creating an untrusted
+   pod
+   <https://github.com/kata-containers/documentation/blob/master/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md#create-an-untrusted-pod-using-kata-containers>`_
+   for more information.
 
-       sudo systemctl daemon-reload
-       sudo systemctl restart crio
+#. Once the cluster initialization is complete, continue reading about how to
+   `Use your cluster`_.
 
 
+CRI+O
+-----
 
-#.  Initialize the master control plane with the command below and follow the displayed instructions to set up `kubectl`.  The :command:`init` command initializes a Kubernetes `control-plane node`_. If you will be adding a pod network add-on, you will need to add configuration to the :command:`init` command line. see the `Install pod network add-on`_ section for details.
+`CRI+O <https://cri-o.io/>`_ is a popular runtime choice because it is
+lightweight and flexible. CRI+O can use OCI compliant runtimes, such as runc
+and kata, and allows specifying runtimes per-Kubernetes service.
 
-    .. code-block:: bash
+This section describes how to configure and run Kubernetes with CRI-O and
+kata-runtime.
+
+#. Start the CRI-O service and enable it to run at boot automatically:
+
+   .. code-block:: bash
+
+      sudo systemctl enable --now crio.service
+
+#. Run the kubeadm command to initialize the master node with the
+   :command:`--cri-socket` parameter:
+
+   .. important:: 
+
+      You may need to add additional parameters to the command below,
+      depending the pod network addon in use. In this example, *flannel* is
+      being used for the pod networking. See `Choose a pod network add-on`_
+      for more information.
+
+   .. code-block:: bash
+
+      sudo kubeadm init \
+      --cri-socket=/run/crio/crio.sock \ #required for CRI+O
+      --pod-network-cidr 10.244.0.0/16   #required for flannel
```

Comment:
This is confusing. It looks like it is all one command.  I'd make it two mini sections with different commands and follow the same format you would use for docker and containerd.

Reply:
It is one big command with in-line comment. Multiple code-blocks are even messier.

Reply:
I'll also add that it renders nicely with the CSS of the website.

Reply:
OK It was how I was seeing it.   This is why for me personally I like seeing multiple init commands for different features. I know it makes the doc a bit harder to read in some cases but it helps me understand more quickly how these commands work.